parsers/mysql/mysql.go

```diff
@@ -99,26 +102,54 @@ type SlowQuery struct {
 	NormalizedQuery string    `json:"normalized_query,omitempty"`
 	DB              string    `json:"db,omitempty"`
 	HostedOn        string    `json:"hosted_on,omitempty"`
+	ReadOnly        bool      `json:"read_only,omitempty"`
 	skipQuery       bool
 }
 
 func (p *Parser) Init(options interface{}) error {
 	p.conf = *options.(*Options)
 	p.nower = &RealNower{}
 	if len(p.conf.Host) > 0 {
-		p.hostedOn = getHostedOn(p.conf.Host, p.conf.User, p.conf.Pass)
+		url := fmt.Sprintf("%s:%s@tcp(%s)/", p.conf.User, p.conf.Pass, p.conf.Host)
+		db, err := sql.Open("mysql", url)
+		if err != nil {
+			log.Fatal(err)
+		}
+		p.db = db
 	}
+
+	// update hostedOn and readOnly every <n> seconds
+	go func() {
+		ticker := time.NewTicker(time.Second * time.Duration(p.conf.QueryInterval))
+		for _ = range ticker.C {
+			p.hostedOn = getHostedOn(p.db)
+			p.readOnly = getReadOnly(p.db)
+		}
+	}()
+
 	return nil
 }
 
-func getHostedOn(host, user, pass string) string {
-	url := fmt.Sprintf("%s:%s@tcp(%s)/", user, pass, host)
-	db, err := sql.Open("mysql", url)
+func getReadOnly(db *sql.DB) bool {
+	rows, err := db.Query("SELECT @@global.read_only;")
 	if err != nil {
 		log.Fatal(err)
 	}
-	defer db.Close()
+	defer rows.Close()
+
+	var value bool
+
+	rows.Next()
+	if err := rows.Scan(&value); err != nil {
+		log.Fatal(err)
+	}
+	if err = rows.Err(); err != nil {
+		log.Fatal(err)
+	}
+	return value
+}
 
+func getHostedOn(db *sql.DB) string {
 	rows, err := db.Query("SHOW GLOBAL VARIABLES like 'basedir';")
```

Comment:
any reason to not make this `SHOW GLOBAL VARIABLES WHERE Variable_name = 'basedir';`?